Packet++/header/NtpLayer.h

```diff
@@ -0,0 +1,649 @@
+        CHU = ('C') | ('H' << 8) | ('U' << 16),
+        /// HF Radio WWV Ft. Collins, CO
+        WWV = ('W') | ('W' << 8) | ('V' << 16),
+        /// HF Radio WWVH Kauai, HI
+        WWVH = ('W') | ('W' << 8) | ('V' << 16) | ('H' << 24),
+        /// NIST telephone modem
+        NIST = ('N') | ('I' << 8) | ('S' << 16) | ('T' << 24),
+        /// NIST telephone modem
+        ACTS = ('A') | ('C' << 8) | ('T' << 16) | ('S' << 24),
+        /// USNO telephone modem
+        USNO = ('U') | ('S' << 8) | ('N' << 16) | ('O' << 24),
+        /// European telephone modem
+        PTB = ('P') | ('T' << 8) | ('B' << 16),
+        /// Meinberg DCF77 with amplitud modulation (Ref: https://www.meinbergglobal.com/english/info/ntp-refid.htm)
+        DCFa = ('D') | ('C' << 8) | ('F' << 16) | ('a' << 24),
+        /// Meinberg DCF77 with phase modulation)/pseudo random phase modulation (Ref: https://www.meinbergglobal.com/english/info/ntp-refid.htm)
+        DCFp = ('D') | ('C' << 8) | ('F' << 16) | ('p' << 24),
+        /// Meinberg GPS (with shared memory access) (Ref: https://www.meinbergglobal.com/english/info/ntp-refid.htm)
+        GPSs = ('G') | ('P' << 8) | ('S' << 16) | ('s' << 24),
+        /// Meinberg GPS (with interrupt based access) (Ref: https://www.meinbergglobal.com/english/info/ntp-refid.htm)
+        GPSi = ('G') | ('P' << 8) | ('S' << 16) | ('i' << 24),
+        /// Meinberg GPS/GLONASS (with shared memory access) (Ref: https://www.meinbergglobal.com/english/info/ntp-refid.htm)
+        GLNs = ('G') | ('L' << 8) | ('N' << 16) | ('s' << 24),
+        /// Meinberg GPS/GLONASS (with interrupt based access) (Ref: https://www.meinbergglobal.com/english/info/ntp-refid.htm)
+        GLNi = ('G') | ('L' << 8) | ('N' << 16) | ('i' << 24),
+        /// Meinberg Undisciplined local clock (Ref: https://www.meinbergglobal.com/english/info/ntp-refid.htm)
+        LCL = ('L') | ('C' << 8) | ('L' << 16),
+        /// Meinberg Undisciplined local clock (Ref: https://www.meinbergglobal.com/english/info/ntp-refid.htm)
+        LOCL = ('L') | ('O' << 8) | ('C' << 16) | ('L' << 24),
+
+        // NTPv3
+
+        /// DCN routing protocol
+        DCN = ('D') | ('C' << 8) | ('N' << 16),
+        /// TSP time protocol
+        TSP = ('T') | ('S' << 8) | ('P' << 16),
+        /// Digital Time Service
+        DTS = ('D') | ('T' << 8) | ('S' << 16),
+        /// Atomic clock (calibrated)
+        ATOM = ('A') | ('T' << 8) | ('O' << 16) | ('M' << 24),
+        /// VLF radio (OMEGA, etc.)
+        VLF = ('V') | ('L' << 8) | ('F' << 16)
+
+    };
+
+    /**
+	 * @class NtpLayer
+	 * Represents a NTP (Network Time Protocol) layer
+	 */
+    class NtpLayer : public Layer
+    {
+    private:
+        ntp_header *getNtpHeader() const { return (ntp_header *)m_Data; }
+
+    public:
+        /**
+		 * A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data (will be casted to @ref ntphdr)
```

Comment:
Now that `ntp_header` is private we need to remove this part: `(will be casted to @ref ntp_header)` because:
1. The user doesn't know what is `ntp_header` (this struct isn't exposed to him)
2. Doxygen gives an error because this struct is private

Reply:
We still need to fix this (doxygen give an error)